File: Concert.java
//package cses_sorting_and_searching;
import java.io.*;
import java.util.*;
 
public class Concert {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		String[] temp = br.readLine().split(" ");
		int n = Integer.parseInt(temp[0]);
		int m = Integer.parseInt(temp[1]);
		
		ArrayList<Integer> tx = new ArrayList<>();
		
		temp = br.readLine().split(" ");
		for(int i = 0; i < n; i++) {
			tx.add(Integer.parseInt(temp[i]));
		}
		
		Collections.sort(tx);
		
		temp = br.readLine().split(" ");
		int p;
		int q;
		for(int i = 0; i < m; i++) {
			p = Integer.parseInt(temp[i]);
			
			q = Collections.binarySearch(tx, p);
			//System.out.println(q + " q");
			
			if(q == -1) {
				System.out.println(-1);
				continue;
			}
			
			q = q < 0 ? (q+1) * -1 - 1: q;
			
			System.out.println(tx.get(q));
			tx.remove(q);
			
			
		}
		
	}
}

File: concert.py
import bisect

def main():
    temp = input().split(" ")
    n = int(temp[0])
    m = int(temp[1])

    tx = []
    line = input().split(" ")
    line2 = input().split(" ")

    for x in range(n):
        tx.append(int(line[x]))
    tx = sorted(tx)

    for x in range(m):
        p = int(line2[x])
        q = bisect.bisect(tx, p)
        #print(q)
        if q == 0:
            print(-1)
        else:
            print(tx[q-1])
            tx.remove(q-1)



if __name__ == '__main__':
    main()
